### PR TITLE
add role class to pull role from auth server

### PR DESCRIPTION
**Pull request by jspies**
Adds the role class so we can query it from admin

One simple endpoint, "mine" returns the logged in user's role.

### CODE REVIEW

File: lib/bypass_auth/role.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+module BypassAuth
+  class Role < Base
+    attr_accessor :name, :role_definition
+
+    def self.mine
+      response = BypassAuth::Client.get("/roles/mine", {}, {})
+      if response.status == 200
+        new(JSON.parse(response.body))
```
Comment (texasjusticar) [inline lib/bypass_auth/role.rb]: does the auth server send back a root node?

Comment (jspies) [inline lib/bypass_auth/role.rb]: It does not on individual items, only on indexes.

**Comment by texasjusticar** [comment]
:shipit:

**Comment by bbean86** [comment]
:shipit:

**Pull request merged**